### PR TITLE
Add codes to framework errors that lack one

### PR DESCRIPTION
**Pull request by lilsweetcaligula**
Fixes: https://github.com/fastify/fastify/issues/4605

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included - (i.e. updated breaking tests)
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/validation.js [hunk 1]
```diff
@@ -134,14 +134,18 @@ function validate (context, request) {
 }
 
 function wrapValidationError (result, dataVar, schemaErrorFormatter) {
+  const FST_ERR_VALIDATION_CODE = 'FST_ERR_VALIDATION'
```
Comment (Uzlopak) [inline lib/validation.js:137]: Is putting it into a variable faster?

Comment (lilsweetcaligula) [reply lib/validation.js]: @Uzlopak You are right. Hardcoding the values is on average faster. I have updated the draft, commit https://github.com/fastify/fastify/pull/4632/commits/b4b8aaeb023fb219d6310bba39739279220b54e3. I am attaching output from the 5-minute benchmarks for reference.

```shell
# hardcoded

$ npx concurrently -k -s first "node ./examples/benchmark/simple.js" "npx autocannon -c 100 -d 300 -p 10 localhost:3000/"
[1] Running 300s test @ http://localhost:3000/
[1] 100 connections with 10 pipelining factor
[1]
[1]
[1] ┌─────────┬──────┬───────┬───────┬───────┬─────────┬─────────┬────────┐
[1] │ Stat    │ 2.5% │ 50%   │ 97.5% │ 99%   │ Avg     │ Stdev   │ Max    │
[1] ├─────────┼──────┼───────┼───────┼───────┼─────────┼─────────┼────────┤
[1] │ Latency │ 5 ms │ 13 ms │ 21 ms │ 24 ms │ 13.1 ms │ 4.33 ms │ 143 ms │
[1] └─────────┴──────┴───────┴───────┴───────┴─────────┴─────────┴────────┘
[1] ┌───────────┬─────────┬─────────┬─────────┬─────────┬──────────┬─────────┬─────────┐
[1] │ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%   │ Avg      │ Stdev   │ Min     │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Req/Sec   │ 70271   │ 72767   │ 73791   │ 75327   │ 73675.95 │ 1915.57 │ 45320   │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Bytes/Sec │ 13.2 MB │ 13.7 MB │ 13.9 MB │ 14.2 MB │ 13.9 MB  │ 360 kB  │ 8.52 MB │
[1] └───────────┴─────────┴─────────┴─────────┴─────────┴──────────┴─────────┴─────────┘
[1]
[1] Req/Bytes counts sampled once per second.
[1] # of samples: 300
[1]
[1] 22104k requests in 300.47s, 4.16 GB read
[1] npx autocannon -c 100 -d 300 -p 10 localhost:3000/ exited with code 0
--> Sending SIGTERM to other processes..
[0] node ./examples/benchmark/simple.js exited with code SIGTERM
```

```shell
# using a variable

$ npx concurrently -k -s first "node ./examples/benchmark/simple.js" "npx autocannon -c 100 -d 300 -p 10 localhost:3000/"
[1] Running 300s test @ http://localhost:3000/
[1] 100 connections with 10 pipelining factor
[1]
[1]
[1] ┌─────────┬──────┬───────┬───────┬───────┬─────────┬─────────┬────────┐
[1] │ Stat    │ 2.5% │ 50%   │ 97.5% │ 99%   │ Avg     │ Stdev   │ Max    │
[1] ├─────────┼──────┼───────┼───────┼───────┼─────────┼─────────┼────────┤
[1] │ Latency │ 6 ms │ 15 ms │ 19 ms │ 21 ms │ 13.4 ms │ 3.94 ms │ 165 ms │
[1] └─────────┴──────┴───────┴───────┴───────┴─────────┴─────────┴────────┘
[1] ┌───────────┬─────────┬─────────┬─────────┬─────────┬──────────┬─────────┬─────────┐
[1] │ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%   │ Avg      │ Stdev   │ Min     │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Req/Sec   │ 69631   │ 70591   │ 72127   │ 73279   │ 72053.23 │ 1775.94 │ 44332   │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼──────────┼─────────┼─────────┤
[1] │ Bytes/Sec │ 13.1 MB │ 13.3 MB │ 13.6 MB │ 13.8 MB │ 13.5 MB  │ 334 kB  │ 8.33 MB │
[1] └───────────┴─────────┴─────────┴─────────┴─────────┴──────────┴─────────┴─────────┘
[1]
[1] Req/Bytes counts sampled once per second.
[1] # of samples: 300
[1]
[1] 21617k requests in 300.27s, 4.06 GB read
[1] npx autocannon -c 100 -d 300 -p 10 localhost:3000/ exited with code 0
--> Sending SIGTERM to other processes..
[0] node ./examples/benchmark/simple.js exited with code SIGTERM
```

**Comment by lilsweetcaligula** [comment]
Rebased.

**Comment by lilsweetcaligula** [comment]
@metcoder95 @Eomm Could you please take a look as well?

File: fastify.js [hunk 2]
```diff
@@ -708,7 +708,7 @@ function fastify (options) {
           const reply = new Reply(res, request, childLogger)
           return frameworkErrors(new FST_ERR_ASYNC_CONSTRAINT(), request, reply)
         }
-        const body = '{"error":"Internal Server Error","code":"FST_ERR_ASYNC_CONSTRAINT","message":"Unexpected error from async constraint","statusCode":500}'
```
Comment (metcoder95) [inline fastify.js]: Can we revert this change?
The change will need further discussion 🙂

Comment (lilsweetcaligula) [reply fastify.js]: @metcoder95 Certainly. Reverted (i.e. the `code` prop removed), commit https://github.com/fastify/fastify/pull/4632/commits/d1dc335c7048620caa8267d74562141b67e0b09c

**Comment by lilsweetcaligula** [comment]
@metcoder95 Because the discussion on internal server errors seems to have stalled: https://github.com/fastify/fastify/issues/4605#issuecomment-1475857299 - should we de-scope internal server errors out of this PR?

**Comment by metcoder95** [comment]
>should we de-scope internal server errors out of this PR?

Let's do that for now. If agreed upon and you're still interested, you can always open a new PR to address them in the future

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by lilsweetcaligula** [comment]
@metcoder95 
> you're still interested, you can always open a new PR to address them in the future

Sure. Also, in the future you may just ping me - and if I have the time and energy (especially the latter) I'll gladly respond.

**Review by metcoder95 — APPROVED** [review]

**Review by Eomm — APPROVED** [review]
